### PR TITLE
6 feature gpa class

### PR DESCRIPTION
**Pull request by vaishnav8791**

### CODE REVIEW

File: src/main/java/Calculations/GPACalculation.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            return 0.7;
+        }else {
+            return 0.0;
+        }
+    }
+
+    /**
+     * Takes list of double consisting of course marks in terms of percentage and iterate through the list and
+     * converting values in terms of GPA using helper function of computationGPA. New converted values added to new
+     * list called gpalist.
```
Comment (MrSunshineVS) [inline src/main/java/Calculations/GPACalculation.java:53]: Documentation is to explicit about function's implementation. Consider rewriting documentation in terms of _what_ the function does rather than _how_.

File: src/main/java/Calculations/GPACalculation.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
+package Calculations;
```
Comment (MrSunshineVS) [inline src/main/java/Calculations/GPACalculation.java:1]: This class lacks a set of tests. Refer to the src/test folder on how tests are written and implemented and try and write a comprehensive test suite that captures edge cases for each function. This is especially important as this is a math heavy class.

**Review by AlanC12138 — DISMISSED** [review]
looked over everything, seems good to me. Nice work

File: src/main/java/Calculations/GPACalculation.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            return 1.7;
+        }else if (percentage >= 57.0){
+            return 1.3;
+        }else if (percentage >= 53.0){
+            return 1.0;
+        }else if (percentage >= 50.0){
+            return 0.7;
+        }else {
+            return 0.0;
+        }
```
Comment (MrSunshineVS) [inline src/main/java/Calculations/GPACalculation.java:47]: Not a pressing issue, but consider refactoring this function to use some kind of abstract gpa evaluation system. As it is written, this function would be hard to extend to other schools and institutions and violates the open/closed principle.

File: src/main/java/Calculations/GPACalculation.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        //System.out.println(gradelist);
+        return gradelist;
+    }
+
+    /**
+     * Takes a list of double of markpercentage (course marks) and a list of courseCredit (0.5 or 1 depending on
+     * half-year or full-year). Computes overall GPA accounting for course credit. GPA is calculated by first
+     * taking sum of credits, then converting markpercentage into gpalist, then creating a new list consisting of
+     * gpa x course credit, and taking sum of this and dividing by sum of credits.
+     *
```
Comment (MrSunshineVS) [inline src/main/java/Calculations/GPACalculation.java:139]: similar issue with documentation (too much explanation of implementation).

File: src/main/java/Calculations/GPACalculation.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+        }
+        for(int s = 0; s < gpalist.size(); s ++){
+            overallGrade.add(gpalist.get(s) * courseCredits.get(s));
+        }
+        for(int v = 0; v < overallGrade.size(); v++) {
+            sumGPA += overallGrade.get(v);
+        }
+        totalGpa = Math.round((sumGPA / sumCredits) * 100.0) / 100.0;
+        //System.out.println(totalGpa);
+        return totalGpa;
```
Comment (MrSunshineVS) [inline src/main/java/Calculations/GPACalculation.java:165]: I don't believe that this function is doing its computations correctly. Again, consider writing a test suite to ensure everything is working properly.

File: src/main/java/Calculations/GPACalculation.java [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+            return "C-";
+        }else if(percentage1 >= 57){
+            return "D+";
+        }else if(percentage1 >= 55){
+            return "D";
+        }else if(percentage1 >= 50){
+            return "D-";
+        }else {
+            return "F";
+        }
```
Comment (MrSunshineVS) [inline src/main/java/Calculations/GPACalculation.java:116]: Similar issue to computationgpa

File: src/main/java/Calculations/GPACalculation.java [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+public class GPACalculation {
+
+    /**
+     * Takes a double of mark in terms of percentage and converts to double in terms of GPA point scale
+     * @param percentage is a double that represents the mark in a course
+     * @return a double of GPA value
+     */
+
+    public static double computationGPA(double percentage){
```
Comment (MrSunshineVS) [inline src/main/java/Calculations/GPACalculation.java:22]: consider renaming to percentageToGPA to comply with other naming decision (percentageToGrade)

**Review by MrSunshineVS — CHANGES_REQUESTED** [review]
Great work! The functions make sense, however I think that this branch needs a set of tests to ensure that all the code is working properly, check out my review for more detail. Also both the computationGPA and percentToGrade functions are very rigid in their implementation and would be difficult to extend/modify. consider creating a separate data class that contains instructions on how to convert. Other than that great stuff!

**Review by berrythebear — DISMISSED** [review]
Good work! Maybe add spaces between if, while, etc., and the parameter brackets

**Review by YuweiCai2003 — DISMISSED** [review]

**Review by MrSunshineVS — APPROVED** [review]

**Review by samielsabri — APPROVED** [review]
looks good!

**Review by YuweiCai2003 — APPROVED** [review]
Great work
